include/boost/geometry/algorithms/detail/overlay/traversal.hpp

```diff
@@ -606,13 +602,25 @@ public :
             return 0;
         }
 
-        return ranked_point.turn_index == start_turn_index
-                && ranked_point.operation_index == start_op_index ? 3
-            : ranked_point.turn_index == start_turn_index ? 2
-            : 1
+        bool const to_start = ranked_point.turn_index == start_turn_index;
+        bool const to_start_index = ranked_point.operation_index == start_op_index;
+
+        bool const next_in_same_cluster
+                = cluster_indices.count(op.enriched.get_next_turn_index()) > 0;
+
+        return to_start && to_start_index ? 4
```

Comment:
+1 for using `enum` here

Reply:
It's now a `priority`, see conversation with Adam. I assume that's OK for you too.